cli/azd/cmd/init.go

```diff
@@ -324,5 +332,18 @@ func (i *initAction) Run(ctx context.Context) error {
 		return fmt.Errorf("saving default environment: %w", err)
 	}
 
+	// If the configuration is empty, set default subscription & location
+	// This will be the case for first run experience
+	if i.config.IsEmpty() {
```

Comment:
Is checking "empty" conceptually right? If we end up adding a different config setting, the logic for setting default sub/location would be broken.

I think we might instead want to check if each specific default key is set or empty

Reply:
In our design discussion we talked about setting these defaults only on first run experience when the user does not already have any configuration set.

Reply:
I've added a `HasDefaults()` func on the account manager that can be used to check if any account related defaults have previously been set.  If not, then we will set values used in `azd init` command.

This also allowed me to remove `Config` from being injected in the command but still won't save us any time since Config is required by the account manager.